### PR TITLE
Feature/rec 24

### PR DESCRIPTION
**Pull request by matthinc**

### CODE REVIEW

File: frontend/src/components/Time.svelte [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  
+</script>
+
+<div>
+  <Card variant="outlined" class="p-2 mt-2 mb-2">
+    <h6>{label}</h6>
+    <Container>
+      <Row>
+        <Col md="6">
+        <Textfield label="Hours" bind:value={hours} class="fullwidth" type="number" />
```
Comment (bremade) [inline frontend/src/components/Time.svelte:27]: Textfields can be negative

File: frontend/src/pages/recipe/new.svelte [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                    </Col>
+                  </Row>
+                </Container>
+              {/each}
+              <Button on:click={addStep}>Add Step</Button>
+              <!-- Tags -->
+            {:else if current_step === 4}
+              <div>
+                <div class="mb-2">Write tag-name and press 'RETURN'</div>
+                <Tags onlyUnique={true} placeholder="Tags" tags={recipe.tags} autoComplete={allTags} />
```
Comment (bremade) [inline frontend/src/pages/recipe/new.svelte:173]: For my taste the deep black as background color is a little bit to much. Would change it to our material theme color #2196f3

File: frontend/src/pages/recipe/new.svelte [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+              <Table>
+                <thead>
+                  <tr>
+                    <th>Ingredient</th><th>Amount</th><th>Unit</th><th></th>
+                  </tr>
+                </thead>
+                {#each recipe.ingredients as ingredient, index}
+                  <tr>
+                    <td><Textfield bind:value={ingredient.name} class="fullwidth"/></td>
+                    <td><Textfield bind:value={ingredient.amount} class="fullwidth"/></td>
```
Comment (bremade) [inline frontend/src/pages/recipe/new.svelte:134]: Amount can be negative

File: frontend/src/pages/recipe/new.svelte [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+              {/each}
+              <Button on:click={addStep}>Add Step</Button>
+              <!-- Tags -->
+            {:else if current_step === 4}
+              <div>
+                <div class="mb-2">Write tag-name and press 'RETURN'</div>
+                <Tags onlyUnique={true} placeholder="Tags" tags={recipe.tags} autoComplete={allTags} />
+              </div>
+              <!-- Publish -->
+            {:else if current_step === 5}
```
Comment (bremade) [inline frontend/src/pages/recipe/new.svelte:176]: Do we realy need an extra step for that? We could add the finish button to the tags step and then add an extra popup to confirm the creation.

**Review by bremade — CHANGES_REQUESTED** [review]
For the most parts great job. I realy like the step system :thumbsup:
What is still missing in general tho are input checks. I was able to create a recipe with empty ingredients and empty steps etc. So either we should add default values or check the input and give direct feedback.

File: frontend/src/pages/recipe/new.svelte (hunk 4 above)
Comment (matthinc) [reply frontend/src/pages/recipe/new.svelte:176]: I actually like this from an UX point of view.
You can go back to a step, change something and then go publish-step and publish the recipe.
I also plan to add some more stuff in this step (maybe a preview or something like that).

**Review by bremade — APPROVED** [review]
LGTM

**Pull request merged**